test/dns/query.cpp

```diff
@@ -99,6 +104,18 @@ TEST_CASE("throw error while fails evdns_set_options for attempts") {
     REQUIRE(false);
 }
 
+TEST_CASE("throw error while fails evdns_set_options for timeout") {
+    try {
+        create_evdns_base<::evdns_base_new, ::evdns_base_nameserver_ip_add,
+                          base_free_evdns_set_options_timeout>(
+            {{"dns/attempts", "nexa"}}, Reactor::global());
+    } catch (std::runtime_error &) {
+        REQUIRE(base_free_evdns_set_options_timeout_flag);
+        return;
+    }
```

Comment:
I think this could be rewritten as:

```REQUIRE_THROWS_AS(( /* statements here surrounded by braces */), <ErrorClass>)```

Could you try to write it like this? I guess the thing that fixes the invocation of the macro is enclosing the statement or statements by another layer of braces. This is a trick I learned recently.

Could you try this solution for here and see whether it works?